cmd/format-xl.go

```diff
@@ -357,7 +362,12 @@ func saveFormatXL(disk StorageAPI, format interface{}) error {
 	}
 
 	// Rename file `uuid.json` --> `format.json`.
-	return disk.RenameFile(minioMetaBucket, tmpFormat, minioMetaBucket, formatConfigFile)
+	if err = disk.RenameFile(minioMetaBucket, tmpFormat, minioMetaBucket, formatConfigFile); err != nil {
+		return err
+	}
+
+	disk.SetDiskID(diskID)
```

Comment:
I think it is better to move this next to saveFormatXL() call and not inside this latter

Reply:
It is on purpose @vadmeste this is to ensure that once the format is saved, the disk is online.